pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -0,0 +1,319 @@
+
+	return rbac, nil
+}
+
+// convertToPermission converts a single AccessRule to a Permission.
+func convertToPermission(rule *rbacproto.AccessRule) *policy.Permission {
+	permission := &policy.Permission{}
+
+	if len(rule.Methods) > 0 {
+		permission.Methods = make([]string, len(rule.Methods))
+		copy(permission.Methods, rule.Methods)
+	}
+
+	if len(rule.Paths) > 0 {
+		permission.Paths = make([]*envoy_type.StringMatch, 0)
+		for _, path := range rule.Paths {
+			permission.Paths = append(permission.Paths, convertToStringMatch(path))
+		}
+	}
+
+	if len(rule.Constraints) > 0 {
+		conditions := make([]*policy.Permission_Condition, 0)
+		for _, v := range rule.Constraints {
+			conditions = append(conditions, &policy.Permission_Condition{
+				ConditionSpec: &policy.Permission_Condition_Header{
+					Header: &policy.MapEntryMatch{
+						Key:    v.Key,
+						Values: convertToStringMatches(v.Values)},
+				}})
+		}
+		permission.Conditions = conditions
+	}
+
+	return permission
+}
+
+func convertToStringMatches(list []string) []*envoy_type.StringMatch {
+	matches := make([]*envoy_type.StringMatch, 0)
+	for _, s := range list {
+		matches = append(matches, convertToStringMatch(s))
+	}
+	return matches
+}
+
+// convertToStringMatch converts a string to a StringMatch, it supports four types of conversion:
+// 1. Exact match, e.g. "abc" is converted to a simple exact match of "abc"
+// 2. Suffix match, e.g. "*abc" is converted to a suffix match of "abc"
+// 3. Prefix match, e.g. "abc* " is converted to a prefix match of "abc"
+// 4. All match. i.e. "*" is converted to a regular expression match of "*"
+func convertToStringMatch(s string) *envoy_type.StringMatch {
+	s = strings.TrimSpace(s)
+	switch {
+	case s == "*":
+		return &envoy_type.StringMatch{MatchPattern: &envoy_type.StringMatch_Regex{Regex: "*"}}
+	case strings.HasPrefix(s, "*"):
+		return &envoy_type.StringMatch{MatchPattern: &envoy_type.StringMatch_Suffix{Suffix: strings.TrimPrefix(s, "*")}}
+	case strings.HasSuffix(s, "*"):
+		return &envoy_type.StringMatch{MatchPattern: &envoy_type.StringMatch_Prefix{Prefix: strings.TrimSuffix(s, "*")}}
+	default:
+		return &envoy_type.StringMatch{MatchPattern: &envoy_type.StringMatch_Simple{Simple: s}}
+	}
+}
+
+// convertToPrincipals converts a list of subjects to principals.
+func convertToPrincipals(bindings []*rbacproto.ServiceRoleBinding) []*policy.Principal {
+	principals := make([]*policy.Principal, 0)
+	for _, binding := range bindings {
+		for _, subject := range binding.Subjects {
+			principals = append(principals, convertToPrincipal(subject))
+		}
+	}
+	return principals
+}
+
+// convertToPrincipal converts a single subject to principal.
+func convertToPrincipal(subject *rbacproto.Subject) *policy.Principal {
+	principal := &policy.Principal{}
+
+	if subject.Group != "" {
+		log.Errorf("group is not supported for now but set to %s.", subject.Group)
+	}
+	attributes := []*policy.Principal_Attribute{}
+	for k, v := range subject.Properties {
```

Comment:
We cannot do it for every property. Since we are only filling out "header" attribute here. Only if property key is "request.headers", we then fill out the "header".

Reply:
Updated to check the actual contents in every property, let me know if the attribute name (`source.ip`, `source.service`, `request.header`, `destination.ip`, `destionation.port`) used here needs any updates.